src/main/java/org/jitsi/jigasi/transcription/TranscriptionResult.java

```diff
@@ -17,85 +17,167 @@
+     * @param participant the participant whose audio was transcribed.
+     * @param alternative the single alternative transcription to add.
      */
-    public TranscriptionResult(String transcription, String name)
+    public TranscriptionResult(
+        Participant participant,
+        TranscriptionAlternative alternative)
     {
-        this.transcription = transcription;
-        this.name = name;
+        this.participant = participant;
+        alternatives.add(alternative);
     }
 
     /**
-     * The transcription of the audio in the request
+     * Create a TranscriptionResult of a TranscriptionRequest which
+     * will hold the text of the audio
      *
-     * @return the transcription as a String
+     * @param alternative the single alternative transcription to add.
      */
-    public String getTranscription()
+    public TranscriptionResult(TranscriptionAlternative alternative)
     {
-        return transcription;
+        participant = null;
+        alternatives.add(alternative);
     }
 
     /**
-     * Get the name of the participant saying the audio
+     * Create a TranscriptionResult of a TranscriptionRequest which
+     * will hold the text of the audio
      *
-     * @return the name
+     * @param alternatives the alternative transcriptions.
      */
-    public String getName()
+    public TranscriptionResult(
+        Collection<TranscriptionAlternative> alternatives)
     {
-        return name;
+        this(null, alternatives);
     }
 
     /**
-     * Set the name of the participant who said this piece of transcription
+     * Create a TranscriptionResult of a TranscriptionRequest which
+     * will hold the text of the audio
      *
-     * @param name the name
+     * @param participant the participant whose audio was transcribed.
      */
-    public void setName(String name)
+    public TranscriptionResult(Participant participant)
     {
-        this.name = name;
+        this.participant = participant;
     }
 
+    /**
+     * Create a TranscriptionResult of a TranscriptionRequest which
+     * will hold the text of the audio
+     */
+    public TranscriptionResult()
+    {
+        participant = null;
+    }
+
+    /**
+     * Adds an alternative transcription to this {@link TranscriptionResult}.
+     * @param alternative the alternative.
+     */
+    public void addAlternative(TranscriptionAlternative alternative)
+    {
+        alternatives.add(alternative);
+    }
+
+    /**
+     * @return all alternative transcription of this {@link TranscriptionResult}.
+     */
+    public Collection<TranscriptionAlternative> getAlternatives()
+    {
+        return alternatives;
+    }
+
+    /**
+     * Get the name of the participant saying the audio
+     *
+     * @return the name
+     */
+    public String getName()
+    {
+        return participant == null
+            ? Participant.UNKNOWN_NAME
+            : participant.getName();
+    }
+
```

Comment:
When rebasing, there should be a method getId(). This should now use the participant. Also, 
in getName(), should UNKNOWN_NAME be changed to the id?